doc/monitoring-stack-deployment/README.md

```diff
@@ -53,11 +63,142 @@ oc expose service prometheus-operated --hostname prometheus.NAMESPACE_NAME.apps.
+metadata: ...
+spec:
+  monitoring:
+    enabled: true # <------ here
+  ...
+```
+
+4. Create new service account for Prometheus
+
+```bash
+oc create serviceaccount prometheus-monitoring
+```
+
+5. Create a ClusterRoleBinding to give Prometheus ServiceAccount the RBAC permissions required to scrape metrics
+
+```bash
+cat << EOF | oc create -f -
+kind: ClusterRoleBinding
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: prometheus-monitoring
+subjects:
+  - kind: ServiceAccount
+    name: prometheus-monitoring
+    namespace: <3scale namespace>
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: cluster-monitoring-view
+EOF
+```
+
+6. Create a token for the prometheus-monitoring ServiceAccount
+
+```bash
+oc create token prometheus-monitoring
+```
+This token will expire, which means the Prometheus will lose access to required resource. You can add the `--duration X<s or m>` to specify how long will the token be valid for. Refer to official OpenShift documentation on managing token lifetime.
+
+7. Update the file `3scale-scrape-configs.yaml` bearer_token field with the token generated from the above.
+
+8. Create additional-scrape-config secret:
+
+```bash
+oc create secret generic additional-scrape-configs --from-file=3scale-scrape-configs.yaml=./3scale-scrape-configs.yaml
+```
+
+9. Deploy prometheus
+
+In `prometheus.yaml` file provided, fill the `spec.externalUrl` field with the external URL. The URL template should be:
+
+```yaml
+spec:
+  ...
+  externalUrl: https://prometheus.NAMESPACE_NAME.apps.CLUSTER_DOMAIN
+```
+
+Then deploy prometheus server:
+
+```bash
+oc apply -f prometheus.yaml
+```
+
+10. Create Prometheus route
+
+```bash
+oc expose service prometheus-operated --hostname prometheus.NAMESPACE_NAME.apps.CLUSTER_DOMAIN
+```
+
+11. Deploy grafana datasource
+
+```bash
+oc apply -f datasource-v5.yaml
+```
+
+12. Deploy grafana
+
+```bash
+oc apply -f grafana-v5.yaml
+```
+
+13. Expose grafana route
+
+```bash
+oc expose service example-grafana-service
 ```
+
+#### Migration from v4 to v5
+
+##### Grafana v4 removal
+
+1. Remove the Grafana custom resource to trigger the deletion of Grafana v4 application along with the service and routes. Double check that the route is removed, if not, remove it manually.
+
+2. Remove the Grafana v4 datasource custom resource
+
+3. Remove the Grafana Operator v4
+
+##### Grafana v5 installation
+
+1. Install Grafana Operator v5
+
+2. Create Grafana v5 datasource (instructions above in `OpenShift versions from 4.16+ and 3scale Operator version 2.15+` section)
+
+3. Create Grafana v5 grafana (instructions above in `OpenShift versions from 4.16+ and 3scale Operator version 2.15+` section)
+
+4. Expose Grafana v5 route (instructions above in `OpenShift versions from 4.16+ and 3scale Operator version 2.15+` section)
+
+5. Create serviceAccount for Prometheus (instructions above in `OpenShift versions from 4.16+ and 3scale Operator version 2.15+` section)
+
```

Comment:
Need to add a step after creating the serviceaccount, token, etc. to restart prometheus

Reply:
Thanks, done